### PR TITLE
Build strategy based on containers with cache

### LINKED ISSUE #213: Optimize Building Strategy Based on Containers
Currently the project uses a multi-stage Dockerfile to build the extension; Although this strategy allows an easy way to reproduce the builds and maintain the versions in sync, it is not aligned with the rest of the extensions.

The other extensions use a single stage build Containerfile and rely the build on the underlying system ( typically a runner within GH Actions).  One of the main reasons to dismiss the multi-stage in favor for a underlaying system build is reuse cache for node modules to avoid wasting time on PR-Checks.

We want to find a mixed solution were the build system can be used on several CI?CD systems where maybe the cache strategy is not available (i.e Tekton)

### PR DESCRIPTION
**Pull request by adrianriobo**
Fix #213

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CONTAINER_MANAGER ?= podman
+
+IMG ?= quay.io/redhat-developer/openshift-local-extension:latest
+IMG_CACHE ?= quay.io/rhqp/openshift-local-extension-cache:latest
+
+# Build the container cache image
+.PHONY: build-cache
+build-cache: 
+	${CONTAINER_MANAGER} build -t ${IMG_CACHE} -f oci/Containerfile.cache .
```
Comment (gbraad) [inline Makefile:9]: Who maintains the lifecycle of this cache?

Comment (adrianriobo) [reply Makefile:9]: the idea would be self managed on every merge to main a gh action will build the image and push it

**Comment by adrianriobo** [comment]
@benoitf added you here as per our discussions about a way to move the cache into a building strategy based on Containerfiles

**Comment by adrianriobo** [comment]
This depends on #222

**Review by benoitf — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Pull request merged**